File: aoj/src/ALDS1_1_A/main.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)
var w = bufio.NewWriter(os.Stdout)

func printArray(a []int) {
	n := len(a)

	fmt.Fprintf(w, "%d", a[0])
	for i := 1; i < n; i++ {
		fmt.Fprintf(w, " %d", a[i])
	}
	fmt.Fprintln(w, "")
}

func insertSort(a []int) {
	n := len(a)

	for i := 1; i < n; i++ {
		printArray(a)
		v := a[i]
		j := i - 1
		for ; j >= 0 && a[j] > v; j-- {
			a[j+1] = a[j]
		}
		a[j+1] = v
	}
	printArray(a)
}

// ALDS1_1_A: 挿入ソート
func main() {
	sc.Buffer([]byte{}, math.MaxInt64)
	sc.Split(bufio.ScanWords)
	defer w.Flush()

	n := nextInt()

	a := make([]int, n, n)
	for i := 0; i < n; i++ {
		a[i] = nextInt()
	}

	insertSort(a)

}

func next() string {
	sc.Scan()
	return sc.Text()
}

func nextInt() int {
	s := next()
	i, _ := strconv.Atoi(s)
	return i
}

File: aoj/src/ALDS1_1_A/main.py
def printarray(x):
    print(" ".join([str(v) for v in x]))

def insertionsort(x):
    printarray(x)
    for i in range(1, len(x)):
        for j in range(i, 0, -1):
            if x[j] > x[j-1]:
                break
            x[j], x[j-1] = x[j-1], x[j]
        printarray(x)

# ALDS1_1_A: 挿入ソート
def main():
    input() # skip scan n
    x = [int(v) for v in input().split(" ")]
    insertionsort(x)

if __name__ == '__main__':
    main()
